c/driver_manager/adbc_driver_manager.cc

```diff
@@ -15,26 +15,41 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#include "arrow-adbc/adbc_driver_manager.h"
+#if defined(_WIN32)
+#pragma comment(lib, "advapi32.lib")  // for registry stuff
+#include <windows.h>                  // Must come first
+
+#include <libloaderapi.h>
+#include <string.h>  // strncasecmp
+#include <strsafe.h>
+
+#ifdef _MSC_VER
+#define strncasecmp _strnicmp
+#endif  // _MSC_VER
+
+#else
+#include <dlfcn.h>
+#endif  // defined(_WIN32)
+
+#include <toml++/toml.hpp>
 #include "arrow-adbc/adbc.h"
+#include "arrow-adbc/adbc_driver_manager.h"
 
 #include <algorithm>
 #include <array>
 #include <cctype>
 #include <cerrno>
 #include <cstring>
+#include <filesystem>  // NOLINT [build/c++17]
 #include <string>
 #include <unordered_map>
 #include <utility>
+#include <vector>
 
-#if defined(_WIN32)
-#include <windows.h>  // Must come first
+using namespace std::string_literals;  // NOLINT [build/namespaces]
 
-#include <libloaderapi.h>
-#include <strsafe.h>
-#else
-#include <dlfcn.h>
-#endif  // defined(_WIN32)
+ADBC_EXPORT
+std::vector<std::filesystem::path> AdbcParsePath(const std::string_view& path);
```

Comment:
Can we use `InternalAdbc` instead?

Reply:
It has to start with `Adbc` because of the symbols.map, so we can't use `InternalAdbc` unless we change the symbols.map 

We could change it to `AdbcInternal` if you like, or just not have the separate unit test for it and not export it, but I prefer to test it

Reply:
Ah, must have missed that. Why doesn't the other internal function use that too then?

Reply:
Right, I'm talking about `AdbcDriverManagerDefaultEntrypoint` which is the other function we only expose for unit testing. It's not using `InternalAdbc` that's why I followed the same pattern. It feels like either both should be `InternalAdbc` or neither should

Reply:
Sure, we can rename it. That probably predates the symbols.map.

Reply:
Sure, I'll rename both then